src/back/glsl.rs

```diff
@@ -119,17 +255,33 @@ pub fn write(module: &Module, out: &mut impl Write) -> Result<(), Error> {
             continue;
         }
 
+        let name = if !es {
+            namer(global.name.as_ref())
+        } else {
+            global.name.clone().ok_or_else(|| {
+                Error::Custom(String::from("Global names must be specified in es"))
+            })?
+        };
+
         if let Some(ref binding) = global.binding {
-            write!(out, "layout({}) ", Binding(binding.clone()))?;
+            if !es {
+                write!(out, "layout({}) ", Binding(binding.clone()))?;
```

Comment:
I'm wondering now... if the GLES doesn't have the layout qualifier, then matching has to happen with names. But names depend on the order they've been encountered. So technically, would we hit a case where the same varying, when exported from VS and FS shaders, will end up with different names? I.e. "myvar4" vs "myvar5"?

We may need to find a more elaborate scheme. For example, see how MSL backend does this.

Reply:
I'm not sure what the problem is but its already required that the name is defined in es